redis/src/cluster_async/mod.rs

```diff
@@ -539,31 +539,31 @@ where
                 let params = params.clone();
                 async move {
                     let result = connect_and_check(&node.0, params, node.1).await;
-                    match result {
-                        Ok(conn) => Some((node.0, async { conn }.boxed().shared())),
-                        Err(e) => {
-                            trace!("Failed to connect to initial node: {:?}", e);
-                            None
-                        }
-                    }
+                    result.map(|conn| (node.0, async { conn }.boxed().shared()))
                 }
             })
             .buffer_unordered(initial_nodes.len())
             .fold(
-                HashMap::with_capacity(initial_nodes.len()),
-                |mut connections: ConnectionMap<C>, conn| async move {
-                    connections.extend(conn);
-                    connections
+                (HashMap::with_capacity(initial_nodes.len()), None),
+                |mut connections: (ConnectionMap<C>, Option<String>), conn| async move {
+                    match conn {
```

Comment:
minor:
`conn` is actually a `Result<(String, ConnectionFuture)>`. The name and the lack of typing hide this from the next programmer. consider using a more informative name.